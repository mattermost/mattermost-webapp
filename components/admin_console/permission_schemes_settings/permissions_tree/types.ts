// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type Permission = {
    id: string;
    combined?: boolean;
    permissions: string[];
}
export type Group = {
    id: string;
    permissions: Array<Permission | string>;
}

export type AdditionalValues = {
    edit_post: {
        editTimeLimitButton: JSX.Element;
    };
}
