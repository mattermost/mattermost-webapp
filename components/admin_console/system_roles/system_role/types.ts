// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type ReadAccess = 'read';
export const readAccess: ReadAccess = 'read';
export type WriteAccess = 'write';
export const writeAccess: WriteAccess = 'write';
export type NoAccess = false;
export const noAccess: NoAccess = false;
export type MixedAccess = 'mixed';
export const mixedAccess: MixedAccess = 'mixed';

export type PermissionAccess = ReadAccess | WriteAccess | NoAccess;
export type PermissionsToUpdate = Record<string, ReadAccess | WriteAccess | NoAccess>;
export type PermissionToUpdate = {
    name: string;
    value: PermissionAccess;
};

export type SystemSection = {
    name: string;
    hasDescription?: boolean;
    subsections?: SystemSection[];
    disabled?: boolean;
}
