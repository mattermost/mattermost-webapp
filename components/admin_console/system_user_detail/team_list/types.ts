// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Team, TeamMembership} from '@mattermost/types/teams';

export type TeamWithMembership = Team & TeamMembership;
