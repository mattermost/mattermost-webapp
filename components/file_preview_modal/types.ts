// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {FileInfo} from '@mattermost/types/files';

export type LinkInfo = {
    has_preview_image: boolean;
    link: string;
    extension: string;
    name: string;
}

export function isFileInfo(info: FileInfo | LinkInfo): info is FileInfo {
    return Boolean((info as FileInfo).id);
}
