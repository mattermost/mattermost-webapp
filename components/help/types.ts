// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum HelpLink {
    Messaging = 'Messaging',
    Composing = 'Composing',
    Mentioning = 'Mentioning',
    Formatting = 'Formatting',
    Attaching = 'Attaching',
    Commands = 'Commands',
}
