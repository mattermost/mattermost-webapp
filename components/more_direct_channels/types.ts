// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Channel} from '@mattermost/types/channels';
import {UserProfile} from '@mattermost/types/users';

import {Value} from 'components/multiselect/multiselect';

// Not to be confused with the GroupChannel type used for LDAP groups
export type GroupChannel = Channel & {
    profiles: UserProfile[];
}

export function isGroupChannel(option: UserProfile | GroupChannel): option is GroupChannel {
    return (option as GroupChannel)?.type === 'G';
}

export type Option = (UserProfile & {last_post_at?: number}) | GroupChannel;

export type OptionValue = Option & Value;

export function optionValue(option: Option): OptionValue {
    return {
        value: option.id,
        label: isGroupChannel(option) ? option.display_name : option.username,
        ...option,
    };
}
