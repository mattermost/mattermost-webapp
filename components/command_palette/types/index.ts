// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum CommandPaletteEntities {
    Channel = 'channel',
    Playbooks = 'playbooks',
    Boards = 'boards',
    Messages = 'messages',
    Files = 'files',
    Runs = 'runs',
    GoTo = 'goTo',
}

export enum ChannelType {
    DM_CHANNEL= 'D',
    GM_CHANNEL= 'G',
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum BoardsType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum PlaybooksType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

