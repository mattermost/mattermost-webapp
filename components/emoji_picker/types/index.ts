// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory, Emoji} from 'mattermost-redux/types/emojis';

export type Category = {
    className: string;
    emojiIds?: string[];
    id: string;
    message: string;
    name: EmojiCategory;
    offset: number;
};

export type Categories = Record<EmojiCategory, Category>;

export type CategoryOrEmojiRow = {
    type: 'categoryHeaderRow' | 'emojisRow';
    row: EmojiCategory | Emoji[];
}
