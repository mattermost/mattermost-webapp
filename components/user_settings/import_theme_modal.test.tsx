// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {Theme} from 'mattermost-redux/types/preferences';

import {mountWithIntl, shallowWithIntl} from 'tests/helpers/intl-test-helper';

import ImportThemeModal from './import_theme_modal';

describe('components/user_settings/ImportThemeModal', () => {
    it('should match snapshot', () => {
        const wrapper = shallowWithIntl(<ImportThemeModal/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should correctly parse a Slack theme', () => {
        const theme: Theme = {
            sidebarBg: '#4d394b',
            sidebarText: '#ffffff',
            sidebarUnreadText: '#ffffff',
            sidebarTextHoverBg: '#3e313c',
            sidebarTextActiveBorder: '#4c9689',
            sidebarTextActiveColor: '#ffffff',
            sidebarHeaderBg: '#3e313c',
            sidebarHeaderTextColor: '#ffffff',
            onlineIndicator: '#38978d',
            awayIndicator: '#ffbc42',
            dndIndicator: '#f74343',
            mentionBg: '#eb4d5c',
            mentionBj: '#ffffff',
            mentionColor: '#145dbf',
            centerChannelBg: '#ffffff',
            centerChannelColor: '#3d3c40',
            newMessageSeparator: '#ff8800',
            linkColor: '#2389d7',
            buttonBg: '#166de0',
            buttonColor: '#ffffff',
            errorTextColor: '#fd5960',
            mentionHighlightBg: '#ffe577',
            mentionHighlightLink: '#166de0',
            codeTheme: 'github',
            type: 'custom',
        };

        const themeString = '#4d394b,#3e313c,#4c9689,#ffffff,#3e313c,#ffffff,#38978d,#eb4d5c';
        const wrapper = mountWithIntl(<ImportThemeModal/>);
        const instance = wrapper.instance();

        const callback = jest.fn();

        instance.setState({show: true, callback});
        wrapper.update();

        wrapper.find('input').simulate('change', {target: {value: themeString}});

        wrapper.find('#submitButton').simulate('click');

        expect(callback).toHaveBeenCalledWith(theme);
    });
});
