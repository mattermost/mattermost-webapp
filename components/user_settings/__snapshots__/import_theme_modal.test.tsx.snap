// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/user_settings/ImportThemeModal should match snapshot 1`] = `
<span>
  <Modal
    animation={true}
    aria-labelledby="importThemeModalLabel"
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogClassName="a11y__modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    role="dialog"
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h1"
        id="importThemeModalLabel"
      >
        <FormattedMessage
          defaultMessage="Import Slack Theme"
          id="user.settings.import_theme.importHeader"
        />
      </ModalTitle>
    </ModalHeader>
    <form
      className="form-horizontal"
      role="form"
    >
      <ModalBody
        bsClass="modal-body"
        componentClass="div"
      >
        <p>
          <FormattedMessage
            defaultMessage="To import a theme, go to a Slack team and look for \\"Preferences -> Sidebar Theme\\". Open the custom theme option, copy the theme color values and paste them here:"
            id="user.settings.import_theme.importBody"
          />
        </p>
        <div
          className="form-group less"
        >
          <div
            className="col-sm-12"
          >
            <input
              className="form-control"
              id="themeVector"
              onChange={[Function]}
              type="text"
              value=""
            />
            <div
              className="input__help"
            />
          </div>
        </div>
      </ModalBody>
      <ModalFooter
        bsClass="modal-footer"
        componentClass="div"
      >
        <button
          className="btn btn-link"
          id="cancelButton"
          onClick={[Function]}
          type="button"
        >
          <FormattedMessage
            defaultMessage="Cancel"
            id="user.settings.import_theme.cancel"
          />
        </button>
        <button
          className="btn btn-primary"
          id="submitButton"
          onClick={[Function]}
          type="submit"
        >
          <FormattedMessage
            defaultMessage="Submit"
            id="user.settings.import_theme.submit"
          />
        </button>
      </ModalFooter>
    </form>
  </Modal>
</span>
`;
